sdk/eventhub/Azure.Messaging.EventHubs/tests/Primitives/EventProcessorTests.cs

```diff
@@ -22,6 +24,947 @@ namespace Azure.Messaging.EventHubs.Tests
     [TestFixture]
     public class EventProcessorTests
     {
+        /// <summary>
+        ///   Provides test cases for the constructor tests.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> ConstructorCreatesDefaultOptionsCases()
+        {
+            var connectionString = "Endpoint=sb://somehost.com;SharedAccessKeyName=ABC;SharedAccessKey=123;EntityPath=somehub";
+            var credential = Mock.Of<TokenCredential>();
+
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString), "connection string with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", connectionString, default(EventProcessorOptions)), "connection string with explicit null options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential), "namespace with default options" };
+            yield return new object[] { new BasicProcessorMock(99, "consumerGroup", "namespace", "hub", credential, default(EventProcessorOptions)), "namespace with explicit null options" };
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(-100)]
+        [TestCase(-10)]
+        [TestCase(-1)]
+        [TestCase(0)]
+        public void ConstructorValidatesTheConsumerGroup(int batchSize)
+        {
+            Assert.That(() => new BasicProcessorMock(batchSize, "dummyGroup", "dummyConnection", new EventProcessorOptions()), Throws.InstanceOf<ArgumentException>(), "The connection string constructor should validate the maximum batch size.");
+            Assert.That(() => new BasicProcessorMock(batchSize, "dummyGroup", "dummyNamespace", "dummyEventHub", Mock.Of<TokenCredential>(), new EventProcessorOptions()), Throws.InstanceOf<ArgumentException>(), "The namespace constructor should validate the maximum batch size.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessor{TPartition}" />
+        ///   constructor.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorValidatesTheConsumerGroup(string consumerGroup)
+        {
+            Assert.That(() => new BasicProcessorMock(1, consumerGroup, "dummyConnection", new EventProcessorOptions()), Throws.InstanceOf<ArgumentException>(), "The connection string constructor should validate the consumer group.");
+            Assert.That(() => new BasicProcessorMock(1, consumerGroup, "dummyNamespace", "dummyEventHub", Mock.Of<TokenCredential>(), new EventProcessorOptions()), Throws.InstanceOf<ArgumentException>(), "The namespace constructor should validate the consumer group.");
+        }
```

Comment:
Maybe for the string parameters we could use that approach that expects an `ArgumentNullException` when the value is `null`, and an `ArgumentException` when the value is empty.

Reply:
I'd call that orthogonal to the responsibility being validated; the Argument class is responsible for ensuring the proper type.  We know that we're using it for validations.  While you could say "what if we forget and do it manually" and you've got a point, but I'd argue that the Argument/ArgumentNull difference doesn't offer enough value to assert against.